midend/interpreter.cpp

```diff
@@ -750,7 +750,7 @@ void ExpressionEvaluator::checkResult(const IR::Expression* expression,
 
 void ExpressionEvaluator::setNonConstant(const IR::Expression* expression) {
     auto type = typeMap->getType(expression, true);
-    if (type->is<IR::BoolLiteral>()) {
+    if (type->is<IR::BoolLiteral>() || type->is<IR::Type_Boolean>()) {
```

Comment:
I would remove `BoolLiteral `here, it does not seem to make sense if you are checking for a NonConstant?

Reply:
Yes, a type cannot be a literal.